xdsl/ir/core.py

```diff
@@ -161,7 +169,19 @@ def replace_by(self, value: SSAValue) -> None:
         # carry over name if possible
         if value.name_hint is None:
             value.name_hint = self.name_hint
-        assert len(self.uses) == 0, "unexpected error in xdsl"
+        assert not self.uses, "unexpected error in xdsl"
+
+    def replace_by_if(self, value: SSAValue, test: Callable[[Use], bool]):
+        """
+        Replace the value by another value in all its uses that pass the given test
+        function.
+        """
+        for use in self.uses.copy():
+            if test(use):
+                use.operation.operands[use.index] = value
+        # carry over name if possible
+        if value.name_hint is None:
```

Comment:
Can we have an option for changing the name?
Compared to `replace_by`, we don't change all uses, so we might run this function multiple times per value, making this name change a bit weird?

Reply:
I doubt we'll ever hit the case you describe, it seems to be primarily useful for "replace everything except" as opposed to some targeted replacing. I'll merge this PR, and happily add an option if you are sure that it would be a good idea